trollmoves/mirror.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2021 Trollmoves developers
+#
+# Author(s):
+#
+#   Martin Raspaud <martin.raspaud@smhi.se>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""All you need for mirroring."""
+
+import os
+
+from six.moves.urllib.parse import urlparse
+
+from trollmoves.client import request_push
+from trollmoves.server import RequestManager, Deleter
+
+# FIXME: don't use globals
+file_registry = {}
+
+
+class MirrorRequestManager(RequestManager):
+    """Handle requests as a mirror."""
+
+    def __init__(self, port, attrs):
+        """Set up this mirror request manager."""
+        RequestManager.__init__(self, port, attrs)
+        self._deleter = MirrorDeleter(attrs)
```

Comment:
The addition of `attrs` to this call is the actual fix and was pretty well hidden among other changes, so I'll just highlight it here.